core/shared/src/main/scala/fs2/concurrent/Topic.scala

```diff
@@ -89,127 +91,52 @@ abstract class Topic[F[_], A] { self =>
       def publish1(b: B): F[Unit] = self.publish1(g(b))
       def subscribe(maxQueued: Int): Stream[F, B] =
         self.subscribe(maxQueued).map(f)
+      def subscribeAwait(maxQueued: Int): Resource[F, Stream[F, B]] =
+        self.subscribeAwait(maxQueued).map(_.map(f))
       def subscribers: Stream[F, Int] = self.subscribers
-      def subscribeSize(maxQueued: Int): Stream[F, (B, Int)] =
-        self.subscribeSize(maxQueued).map { case (a, i) => f(a) -> i }
     }
 }
 
 object Topic {
 
-  /** Constructs a `Topic` for a provided `Concurrent` datatype. The
-    * `initial` value is immediately published.
-    */
-  def apply[F[_], A](initial: A)(implicit F: Concurrent[F]): F[Topic[F, A]] = {
-    implicit def eqInstance: Eq[Strategy.State[A]] =
-      Eq.instance[Strategy.State[A]](_.subscribers.keySet == _.subscribers.keySet)
-
-    PubSub(PubSub.Strategy.Inspectable.strategy(Strategy.boundedSubscribers(initial)))
-      .map { pubSub =>
+  /** Constructs a Topic */
+  def apply[F[_], A](implicit F: Concurrent[F]): F[Topic[F, A]] =
+    F.ref(LongMap.empty[Q[F, A]] -> 1L)
+      .product(SignallingRef[F, Int](0))
+      .map { case (state, subscriberCount) =>
         new Topic[F, A] {
-          def subscriber(size: Int): Stream[F, ((Unique, Int), Stream[F, SizedQueue[A]])] =
-            Stream
-              .bracket(
-                Unique[F]
-                  .tupleRight(size)
-                  .flatTap(selector => pubSub.subscribe(Right(selector)))
-              )(selector => pubSub.unsubscribe(Right(selector)))
-              .map { selector =>
-                selector ->
-                  pubSub.getStream(Right(selector)).flatMap {
-                    case Right(q) => Stream.emit(q)
-                    case Left(_)  => Stream.empty // impossible
-                  }
+          def publish1(a: A): F[Unit] =
+            state.get.flatMap { case (subs, _) =>
+              subs.foldLeft(F.unit) { case (op, (_, q)) =>
+                op >> q.offer(a)
               }
```

Comment:
I may be approaching this from afar, but is this a case of `traverse_`?

Reply:
it is `traverse_`, but there is no `Traverse[LongMap]`